qlib/utils/data.py

```diff
@@ -103,3 +105,19 @@ def update_config(base_config: dict, ext_config: Union[dict, List[dict]]):
                     # one of then are not dict. Then replace
                     base_config[key] = ec[key]
     return base_config
+
+
+def guess_horizon(label):
+    """
+    Try to guess the horizon by parsing label
+    """
+    regex = r"Ref\(\s*\$[a-zA-Z]+,\s*-(\d+)\)"
+    horizon_list = [int(x) for x in re.findall(regex, label)]
+
+    if len(horizon_list) == 0:
+        return 0
+    max_horizon = max(horizon_list)
+    # Unlikely the label doesn't use future information
+    if max_horizon < 2:
+        return 0
+    return max_horizon + 1
```

Comment:
Why is it `return max_horizon+1` here instead of `return max_horizon` or `return max_horizon-min_horizon`?

Reply:
This is consistent with exisiting deinition of "horizon" in other code like DDG-DA and TRA. There is no official documentation, but it means the number of future days should be truncated to avoid leaking future information. Using max_horizon + 1 just align with exisitng code base.